core/multi_index.cpp

```diff
@@ -33,7 +33,11 @@ void validate_bucket_indices_impl(const ElementArrayViewParams &param0,
   for (scipp::index i = 0; i < iterDims.volume(); ++i) {
     const auto [i0, i1] = index.get();
     if (size(indices0[i0]) != size(indices1[i1]))
-      throw except::BucketError("Bucket size mismatch");
+      throw except::BucketError(
```

Comment:
Now we are not using the term Bucket elsewhere do we want to rename this to BinError or BinningError?

Reply:
I had started, but then remembered it will cause merge conflicts with the `bin` PR. Will rename afterwards.